pkgs/misc/vim-plugins/overrides.nix

```
@@ -499,11 +499,12 @@ self: super: {
   # or for all grammars:
   # pkgs.vimPlugins.nvim-treesitter.withPlugins (_: tree-sitter.allGrammars)
   nvim-treesitter = super.nvim-treesitter.overrideAttrs (old: {
-    passthru.withPlugins =
-      grammarFn: self.nvim-treesitter.overrideAttrs (_: {
+    passthru.withPlugins = grammarFn:
+      self.nvim-treesitter.overrideAttrs (_: {
+        patches = lib.optional stdenv.isDarwin
+          ./patches/nvim-treesitter/macos-dyld.patch;
```

shouldn't this be up a level?
```suggestion
    patches = lib.optional stdenv.isDarwin
        ./patches/nvim-treesitter/macos-dyld.patch;
    passthru.withPlugins = grammarFn:
      self.nvim-treesitter.overrideAttrs (_: {

```

---

Well, if one decides not to use `withPlugins`, and instead prefer to have `nvim-treesitter` build its own parsers, then this patch will break it, as `nvim-treesitter`'s build will output `.so` suffixed files.

Thanks!

---

> as nvim-treesitter's build will output .so suffixed files

That seems like an upstream issue. Is it meant to be used on non-linux systems?

---

has upstream fixed the issue yet ?